server/asset/eth/eth.go

```diff
@@ -184,20 +176,20 @@ func (eth *Backend) Connect(ctx context.Context) (*sync.WaitGroup, error) {
 	}
 	eth.node = &c
 
-	// Prime the cache with the best block.
-	bestHash, err := eth.node.bestBlockHash(ctx)
+	// Prime the cache with the best block hash.
+	hdr, err := c.bestHeader(ctx)
 	if err != nil {
-		eth.shutdown()
-		return nil, fmt.Errorf("error getting best block hash from geth: %w", err)
+		return nil, fmt.Errorf("error getting best block header from geth: %w", err)
 	}
-	block, err := eth.node.block(ctx, bestHash)
-	if err != nil {
-		eth.shutdown()
-		return nil, fmt.Errorf("error getting best block from geth: %w", err)
-	}
-	_, err = eth.blockCache.add(block)
-	if err != nil {
-		eth.log.Errorf("error adding new best block to cache: %v", err)
+	eth.hashCache = &hashCache{
+		log:        eth.log.SubLogger("hashcache"),
+		node:       &c,
+		signalMtx:  &eth.signalMtx,
+		blockChans: eth.blockChans,
```

Comment:
We don't still need to hold a read lock for ranging over the `hc.blockChans` map?

Reply:
`hashCache` has it's own mutex, and the use pattern in `poll` as written is that it will always be locked when both `blockChans` and `best` are accessed.  Do you need two mutexes?

Reply:
Yeah, you are right.

So originally this was created with the unconnected wallet, so a lot of the wierdness is to make the exported `BlockChannel` not panic before hashCache is created in `Connect`

Reply:
I think I'll just remove the type since it doesn't really do that much anymore.